### PR TITLE
Added Floyd's cycle detection algorithm in Linked List

### LINKED ISSUE #1078: Cycle detection problem in Linked list in C,C++

### PR DESCRIPTION
**Pull request by Vaibhavraj10**
Fixes #1078 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added C code for Detecting Cycle from a linked list using Floyd's cycle detection algorithm

#### Languages Used:
- C

#### Files Added:
- Floyd's_Cycle_Detection_Algorithm.c

### CODE REVIEW

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+/* Input: 
+   5
+   20 30 40 50 60
+
+  Output:
+  Enter the size of Linked list: 
+  Cycle is found!
+*/  
+
```
Comment (singh-shreya6) [inline Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: Please remove extra blank lines at the end of code.

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    int data;
+    struct node* next;
+}node;
+
+node *head = NULL;
+
+ // Inserting a new node into the Linked List
+node* inserting_a_new_node(int data)
+{
+    node* temp;
```
[the remark this answers is not in the document]
Comment (Vaibhavraj10) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: Made the changes.

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return head;
+}
+
+ // Detect if a cycle is found
+int CycleDetection(node* head)
+{
+    node *pointer_fast = head, *pointer_slow = head;
+    while ( pointer_fast && pointer_fast->next && pointer_slow )
+    {
+        pointer_slow = pointer_slow->next;
```
[the remark this answers is not in the document]
Comment (Vaibhavraj10) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: Removed.

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c (hunk 1 above)
Comment (Vaibhavraj10) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: Removed.

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+   20 30 40 50 60
+
+  Output:
+  Enter the size of Linked list: 
+  Cycle is found!
+*/  
+
+   
+   
+
```
[the remark this answers is not in the document]
Comment (Vaibhavraj10) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: Removed

**Comment by KanchanThareja** [comment]
@Vaibhavraj10 good work
@jainaman224 I approve of this PR. Please take a final look

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+// Using the main method
+int main()
+{
+    int n;
+    printf("Enter the size of Linked list: \n");
+    scanf("%d",&n);
+    int linkedList[n];
+    for(int i = 0; i < n; i++)
+    {
+        scanf("%d",&linkedList[i]);
```
Comment (GOVINDDIXIT) [inline Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: space after the comma, Rest LGTM 👍

Comment (Vaibhavraj10) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection_Algorithm.c]: Made the proposed changes.

**Pull request merged**